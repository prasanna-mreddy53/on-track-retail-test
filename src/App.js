import React from "react";
import "./App.scss";

import Header from "./header/header.component";

import { Route, Switch, BrowserRouter } from "react-router-dom";
import Home from "./home/home.component";
function App() {
  return (
    <BrowserRouter>
      <React.Fragment>
        <Header />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/books/:page" component={Home} />
        </Switch>
      </React.Fragment>
    </BrowserRouter>
  );
}

export default App;
