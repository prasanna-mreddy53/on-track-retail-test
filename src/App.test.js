import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import { Route, Switch } from "react-router-dom";

const app = shallow(<App />);

describe("App Component Tests", () => {
  it("renders app correctly", () => {
    expect(app).toHaveLength(1);
  });

  it("should contain a Switch Router", () => {
    expect(app.find(Switch)).toHaveLength(1);
  });

  it("should contain 2 route components", () => {
    expect(app.find(Route)).toHaveLength(2);
  });
});
