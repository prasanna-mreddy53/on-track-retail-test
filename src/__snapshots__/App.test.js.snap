// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Tests should match snapshot 1`] = `ShallowWrapper {}`;
